### PR TITLE
GC disk related code.

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli and @geirst PR

### CODE REVIEW

File: storage/src/tests/persistence/persistencetestutils.cpp [hunk 1]
```diff
@@ -76,15 +70,14 @@ PersistenceTestUtils::PersistenceTestUtils() = default;
...
     return dynamic_cast<spi::dummy::DummyPersistence&>(_env->_node.getPersistenceProvider()).dumpBucket(makeSpiBucket(bid));
 }
 
 void
-PersistenceTestUtils::setupDisks(uint32_t numDisks) {
-    _env = std::make_unique<PersistenceTestEnvironment>(DiskCount(numDisks), "todo-make-unique-persistencetestutils");
-    setupExecutor(numDisks);
+PersistenceTestUtils::setupDisks() {
+    _env = std::make_unique<PersistenceTestEnvironment>("todo-make-unique-persistencetestutils");
+    setupExecutor(2);
```
Comment (geirst) [inline storage/src/tests/persistence/persistencetestutils.cpp:80]: Should this be 1 instead?

File: storage/src/vespa/storage/persistence/filestorage/filestorhandler.h [hunk 2]
```diff
@@ -108,14 +105,14 @@ class FileStorHandler : public MessageSender {
      * Schedule a storage message to be processed by the given disk
      * @return True if we maanged to schedule operation. False if not
      */
-    bool schedule(const std::shared_ptr<api::StorageMessage>&, uint16_t disk);
+    bool schedule(const std::shared_ptr<api::StorageMessage>&);
 
     /**
      * Used by file stor threads to get their next message to process.
      *
      * @param disk The disk to get messages for
```
Comment (geirst) [inline storage/src/vespa/storage/persistence/filestorage/filestorhandler.h:113]: This should be removed.

**Review by geirst — APPROVED** [review]
👍

File: storage/src/vespa/storage/persistence/filestorage/filestorhandler.h (hunk 2 above)
Comment (baldersheim) [reply storage/src/vespa/storage/persistence/filestorage/filestorhandler.h]: Fixed

File: storage/src/tests/persistence/persistencetestutils.cpp (hunk 1 above)
Comment (baldersheim) [reply storage/src/tests/persistence/persistencetestutils.cpp]: Yes it should, I just tested with 2 too as it did not see the connection between disk and executors.

**Review by vekterli — APPROVED** [review]
👍🎉

File: storage/src/tests/persistence/persistencetestutils.cpp [hunk 3]
```diff
@@ -25,13 +25,11 @@ namespace {
...
 
-    vdstestlib::DirConfig initialize(uint32_t numDisks, const std::string & rootOfRoot) {
+    vdstestlib::DirConfig initialize(const std::string & rootOfRoot) {
         vdstestlib::DirConfig config(getStandardConfig(true, rootOfRoot));
         std::string rootFolder = getRootFolder(config);
         vespalib::rmdir(rootFolder, true);
-        for (uint32_t i = 0; i < numDisks; i++) {
-            vespalib::mkdir(vespalib::make_string("%s/disks/d%d", rootFolder.c_str(), i), true);
-        }
+        vespalib::mkdir(vespalib::make_string("%s/disks/d0", rootFolder.c_str()), true);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply storage/src/tests/persistence/persistencetestutils.cpp:32]: Not that I know of.

File: storage/src/vespa/storage/persistence/filestorage/filestorhandlerimpl.cpp [hunk 4]
```diff
@@ -151,11 +141,9 @@ FileStorHandlerImpl::clearMergeStatus(const document::Bucket& bucket, const api:
...
 FileStorHandlerImpl::flush(bool killPendingMerges)
 {
-    for (uint32_t i=0; i<_diskInfo.size(); ++i) {
-        LOG(debug, "Wait until queues and bucket locks released for disk '%d'", i);
-        _diskInfo[i].flush();
-        LOG(debug, "All queues and bucket locks released for disk '%d'", i);
-    }
+        LOG(debug, "Wait until queues and bucket locks released.");
+        _disk.flush();
+        LOG(debug, "All queues and bucket locks released.");
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply storage/src/vespa/storage/persistence/filestorage/filestorhandlerimpl.cpp:146]: It is on its way in a new PR

File: storage/src/vespa/storage/persistence/filestorage/filestorhandlerimpl.cpp [hunk 5]
```diff
@@ -1295,22 +1232,21 @@ FileStorHandlerImpl::getStatus(std::ostream& out, const framework::HttpUrlPath&
...
-        switch (disk.getState()) {
-            case FileStorHandler::AVAILABLE: out << "AVAILABLE"; break;
-            case FileStorHandler::DISABLED: out << "DISABLED"; break;
-            case FileStorHandler::CLOSED: out << "CLOSED"; break;
-        }
-        out << "<h4>Active operations</h4>\n";
-        disk.dumpActiveHtml(out);
-        if (!verbose) continue;
+
+    out << "<h2>Disk " << "</h2>\n";
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply storage/src/vespa/storage/persistence/filestorage/filestorhandlerimpl.cpp:1236]: Will do in followup PR

**Pull request merged**